content/2021/2021-05/2021-05-24-pulsar-weekly.md

```diff
@@ -0,0 +1,150 @@
+
+- [Pulsar client] Add `org.apache.bookkeeper:cpu-affinity` to the shaded profile.
+
+    https://github.com/apache/pulsar/pull/10681 ([merlimat](https://api.github.com/users/merlimat))
+
+- [BookKeeper] Ensure the `ReadHandle` is properly closed on cache invalidation.
+
+    https://github.com/apache/pulsar/pull/10659 ([merlimat](https://api.github.com/users/merlimat))
+
+- [Broker] Support configuring busy-wait in the Pulsar broker and Pulsar client.
+
+    https://github.com/apache/pulsar/pull/10661 ([merlimat](https://api.github.com/users/merlimat))
+
+- [Broker] Set up transaction metadata with `MetadataStore`.
+
+    https://github.com/apache/pulsar/pull/10677 ([fantapsody](https://api.github.com/users/fantapsody))
+
+- [Broker] Set up initial namespaces with `MetadataStore`.
+
+    https://github.com/apache/pulsar/pull/10612 ([fantapsody](https://api.github.com/users/fantapsody))
+
+- [BookKeeper] Support configuring the number of BookKeeper client worker threads.
+
+    https://github.com/apache/pulsar/pull/10649 ([merlimat](https://api.github.com/users/merlimat))
+
+- [Metrics] Add metrics for the non-contiguous deleted message range.
+
+    https://github.com/apache/pulsar/pull/10638 ([Sunny-Island](https://api.github.com/users/Sunny-Island))
+
+- [Performance] Optimizes the `AuthProviderToken` class to use the same instance of the parser.
+
+    https://github.com/apache/pulsar/pull/10664 ([addisonj](https://api.github.com/users/addisonj))
+
+- [Functions] Process asynchronous results in the same Java runnable thread.
+
+    https://github.com/apache/pulsar/pull/10618 ([sijie](https://api.github.com/users/sijie))
+
+- [Broker] Avoid the context switch when `managedLedgerNewEntriesCheckDelayInMillis` is set to `0`.
+
+    https://github.com/apache/pulsar/pull/10660 ([merlimat](https://api.github.com/users/merlimat))
+
+- [Transaction] Support getting the transaction coordinator status through the transaction Admin API.
+
+    https://github.com/apache/pulsar/pull/10639 ([congbobo184](https://api.github.com/users/congbobo184))
+
+- [Pulsar Admin] Return a better error message.
+
+    https://github.com/apache/pulsar/pull/10628 ([eolivelli](https://api.github.com/users/eolivelli))
+
+- [Java][Schema] Enable disabled tests about `AutoConsume` and `KeyValue` and add the test about null values.
+
+    https://github.com/apache/pulsar/pull/10626 ([eolivelli](https://api.github.com/users/eolivelli))
+
+- [Java][Schema] Add more tests about `KeyValue` and NULL values.
+
+    https://github.com/apache/pulsar/pull/10609 ([eolivelli](https://api.github.com/users/eolivelli))
+
+- [Broker] Set up the Pulsar cluster with `MetadataStore`.
+
+    https://github.com/apache/pulsar/pull/10600 ([fantapsody](https://api.github.com/users/fantapsody))
+
+### Notable Bug Fix
+
+- [Admin] Fix the issue that when getting the partition metadata of a non-existent topic, it returns `0`.
```

Comment:
"when getting the partition metadata of a non-existent topic, it returns `0`" Is this the issue or the solution?

Reply:
it is the issue. there should not `0` is returned.

Reply:
Thank you for your explanation. We can write it more clearly next time.